### PR TITLE
Support responseHeaders and Stream response body

### PR DESCRIPTION
**Pull request by sarialem**

### CODE REVIEW

File: src/FakeServer.ts [hunk 1]
```diff
@@ -128,13 +129,18 @@ export default class FakeServer {
                 self.logger(
                     `fakeServer:: call to [${this.req.method} ${this.url} ${JSON.stringify(
                         this.request.body
-                    )}]. Respond with status: [${firstMatch.statusCode}] and body: [${JSON.stringify(
+                    )}]. Respond with status: [${firstMatch.statusCode}], body: [${JSON.stringify(
                         firstMatch.response
-                    )}]`
+                    )}] and headers [${JSON.stringify(firstMatch.responseHeaders)}]`
```
Comment (noamokman) [inline src/FakeServer.ts]: can you make this optional?

File: http-host/package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
   "dependencies": {
     "body-parser": "^1.18.3",
     "express": "^4.16.3",
-    "simple-fake-server": "^2.1.3",
+    "into-stream": "^5.1.0",
+    "simple-fake-server": "file:../dist",
```
Comment (noamokman) [inline http-host/package.json]: revert

Comment (sarialem) [reply http-host/package.json]: Ugh, good catch

File: http-host/index.js [hunk 3]
```diff
@@ -33,12 +35,17 @@ app.post('/fake_server_admin/calls', ({ body: { method: mockedMethod, url: mocke
...
       .to(mockedUrl);
   }
-  if (statusCode && statusCode !== 200) {
+  if (statusCode && statusCode >= 400) {
     call = mock
       .willFail(statusCode);
   } else {
+    let finalResponse = mockedResponse;
+    if (finalResponse) {
+      finalResponse = isJson ? JSON.parse(finalResponse) : finalResponse;
```
Comment (guysegal) [inline http-host/index.js]: isJson -> responseAsJson

File: src/FakeServer.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ export type MockedCall = {
     queryParamsObject?: {};
     statusCode?: number;
     response?: any;
+    responseHeaders?: {[key: string]: string};
```
Comment (guysegal) [inline src/FakeServer.ts]: you can use Record<string, string> type

File: src/FakeServer.ts [hunk 5]
```diff
@@ -128,13 +129,20 @@ export default class FakeServer {
...
+                    )}]${
+                        firstMatch.responseHeaders ? ` and headers [${JSON.stringify(firstMatch.responseHeaders)}]` : ''
+                    }`
                 );
 
                 this.status = firstMatch.statusCode;
                 this.body = firstMatch.response;
+                const responseObject = this.response;
+                if (firstMatch.responseHeaders) {
+                    const hdrs = firstMatch.responseHeaders;
```
Comment (guysegal) [inline src/FakeServer.ts]: cosmetics: hdrs -> headres

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "simple-fake-server",
-  "version": "2.3.1",
+  "version": "2.3.2",
```
Comment (guysegal) [inline package.json]: feels more like - 2.4.0 (new feature, non breaking)

File: http-host/index.js (hunk 3 above)
Comment (sarialem) [reply http-host/index.js]: I thought about this already, but the same terminology is used in both server and client, changing these would be a breaking change.

File: package.json (hunk 6 above)
Comment (sarialem) [reply package.json]: Agree

File: src/FakeServer.ts (hunk 4 above)
Comment (sarialem) [reply src/FakeServer.ts]: Done

File: src/FakeServer.ts (hunk 5 above)
Comment (sarialem) [reply src/FakeServer.ts]: Done

**Pull request merged**